src/core/geometry/qgswkbtypes.h

```diff
@@ -421,11 +423,155 @@ class CORE_EXPORT QgsWkbTypes
       return Unknown;
     }
 
+
+    /**
+     * Returns the curve type for a WKB type. For example, for Polygon WKB types the multi type would be CurvePolygon.
+     * \see isMultiType()
+     * \see singleType()
+     * \see flatType()
+     * \see multiType()
+     *
+     * \since QGIS 3.10
+     */
+    static Type curveType( Type type )
+    {
+      switch ( type )
+      {
+        case Unknown:
+        case Triangle:
+        case TriangleZ:
+        case TriangleM:
+        case TriangleZM:
+          return Unknown;
+
+        case GeometryCollection:
+          return GeometryCollection;
+
+        case GeometryCollectionZ:
+          return GeometryCollectionZ;
+
+        case GeometryCollectionM:
+          return GeometryCollectionM;
+
+        case GeometryCollectionZM:
+          return GeometryCollectionZM;
+
+        case Point:
+          return Point;
+
+        case MultiPoint:
+          return MultiPoint;
+
+        case PointZ:
+          return PointZ;
+
+        case MultiPointZ:
+          return MultiPointZ;
+
+        case PointM:
+          return PointM;
+
+        case MultiPointM:
+          return MultiPointM;
+
+        case PointZM:
+          return PointZM;
+
+        case MultiPointZM:
+          return MultiPointZM;
+
+        case LineString:
+        case CompoundCurve:
+        case CircularString:
+          return CompoundCurve;
+
+        case MultiLineString:
+        case MultiCurve:
+          return MultiCurve;
+
+        case LineStringZ:
+        case CompoundCurveZ:
+        case CircularStringZ:
+          return CompoundCurveZ;
+
+        case MultiLineStringZ:
+        case MultiCurveZ:
+          return MultiCurveZ;
+
+        case LineStringM:
+        case CompoundCurveM:
+        case CircularStringM:
+          return CompoundCurveM;
+
+        case MultiLineStringM:
+        case MultiCurveM:
+          return MultiCurveM;
+
+        case LineStringZM:
+        case CompoundCurveZM:
+        case CircularStringZM:
+          return CompoundCurveZM;
+
+        case MultiLineStringZM:
+        case MultiCurveZM:
+          return MultiCurveZM;
+
+        case Polygon:
+        case CurvePolygon:
+          return CurvePolygon;
+
+        case MultiPolygon:
+        case MultiSurface:
+          return MultiSurface;
+
+        case PolygonZ:
+        case CurvePolygonZ:
+          return CurvePolygonZ;
+
+        case MultiPolygonZ:
+        case MultiSurfaceZ:
+          return MultiSurfaceZ;
+
+        case PolygonM:
+        case CurvePolygonM:
+          return CurvePolygonM;
+
+        case MultiPolygonM:
+        case MultiSurfaceM:
+          return MultiSurfaceM;
+
+        case PolygonZM:
+        case CurvePolygonZM:
+          return CurvePolygonZM;
+
+        case MultiPolygonZM:
+        case MultiSurfaceZM:
+          return MultiSurfaceZM;
+
+        case NoGeometry:
+          return NoGeometry;
+
+        case Point25D:
+        case MultiPoint25D:
```

Comment:
Hmm, should these be handled like triangles (Unknown) since there's no curve type for these 25D ones available?